### PR TITLE
Tutorial fixes and updates

### PR DESCRIPTION
**Pull request by dafurman**
I just went through the written tutorials - this is such a cool format for that style of teaching!

I found some typos and minor points where clarity could be added, so here's a PR for that!

### CODE REVIEW

File: Sources/ComposableArchitecture/Documentation.docc/Tutorials/MeetTheComposableArchitecture/01-Essentials/03-TestingYourFeatures/01-03-01-code-0001.swift [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import ComposableArchitecture
 import XCTest
 
+@testable import CounterApp
```
Comment (dafurman) [inline Sources/ComposableArchitecture/Documentation.docc/Tutorials/MeetTheComposableArchitecture/01-Essentials/03-TestingYourFeatures/01-03-01-code-0001.swift:4]: Just mimicking what's going on in the Navigation part, where the main app target is imported

File: Sources/ComposableArchitecture/Documentation.docc/Tutorials/MeetTheComposableArchitecture/02-Navigation/04-NavigationStacks/02-04-01-code-0006.swift [hunk 2]
```diff
@@ -6,6 +6,6 @@ struct ContactDetailView: View {
   var body: some View {
     Form {
     }
-    .navigationBarTitle(Text(store.contact.name))
+    .navigationTitle(Text(store.contact.name))
```
Comment (dafurman) [inline Sources/ComposableArchitecture/Documentation.docc/Tutorials/MeetTheComposableArchitecture/02-Navigation/04-NavigationStacks/02-04-01-code-0006.swift:9]: Switching from the deprecated [navigationBarTitle](https://developer.apple.com/documentation/swiftui/view/navigationbartitle(_:)-6p1k7) to the newer modifier.

**Comment by dafurman** [comment]
I didn't quite know how to address it in this PR, but I'm wondering if [this bit](https://github.com/pointfreeco/swift-composable-architecture/blob/3104c9ba193b89f6bea885cce277bcfb7528486b/Sources/ComposableArchitecture/Documentation.docc/Tutorials/MeetTheComposableArchitecture/02-Navigation/01-YourFirstPresentation/02-01-YourFirstPresentation.tutorial#L212) should be rephrased:
> The library comes with a variety of tools that mimic SwiftUI's native navigation tools (such
      as sheets, popovers, fullscreen covers, alerts, and confirmation dialogs), but they take
      ``ComposableArchitecture/Store``s instead of bindings.
      
Given that the example with the sheet is actually just using the native binding there:

Screenshot 2024-04-19 at 11 29 26 AM

File: Sources/ComposableArchitecture/Documentation.docc/Articles/TreeBasedNavigation.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ struct InventoryFeature {
 > Note: The key path used with `ifLet` focuses on the `@PresentationState` projected value since it 
 > uses the `$` syntax. Also note that the action uses a
 > [case path](http://github.com/pointfreeco/swift-case-paths), which is analogous to key paths but
-> tuned for enums, and uses the forward slash syntax.
```
Comment (dafurman) [inline Sources/ComposableArchitecture/Documentation.docc/Articles/TreeBasedNavigation.md:90]: The example above uses the newer backslash syntax 🥳

**Review by stephencelis — APPROVED** [review]
Thanks!

**Comment by stephencelis** [comment]
> I didn't quite know how to address it in this PR, but I'm wondering if [this bit](https://github.com/pointfreeco/swift-composable-architecture/blob/3104c9ba193b89f6bea885cce277bcfb7528486b/Sources/ComposableArchitecture/Documentation.docc/Tutorials/MeetTheComposableArchitecture/02-Navigation/01-YourFirstPresentation/02-01-YourFirstPresentation.tutorial#L212) should be rephrased:

Good call! Will fix it up on `main`.

**Pull request merged**